### DISCUSSION TITLE
Issues are not closed automatically

### DISCUSSION THREAD

**Discussion by TheMostDiligent** [github/codeql#8987] [Q&A]
Hello!
Are issues supposed to be closed automatically when they are fixed?
After merging changes that fixed some code scanning issues, they are labeled as fixed, but still remain open. I can manually dismiss the issues with one of the 3 available options (false positive, won't fix, and used in test), none of which really reflect that the issue was in fact fixed.

**Comment by Crell** [comment]
We're seeing the same behavior as well.  No indication of how to get the issues to resolve.

**Comment by ranma2913** [comment]
Me too. This doesn't make sense that CodeQL can tell it's fixed, but not clear the Code Scanning Alert.

![image](https://github.com/github/codeql/assets/4295880/9979db6f-ff90-437a-aed4-5bb29e1209a9)

**Reply by ranma2913** [reply]
in my case it may be related to renaming the workflow file. I moved to the reusable-workflow pattern. 
See discussion [#23403](https://github.com/orgs/community/discussions/23403)

**Reply by ranma2913** [reply]
Can confirm I was able to clear away the fixed issues with commands like:
[Get a code scanning analysis for a repository](https://docs.github.com/en/rest/code-scanning/code-scanning?apiVersion=2022-11-28#get-a-code-scanning-analysis-for-a-repository)
```shell
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer $credentials_GHEC_TOKEN" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/OWNER/REPO/code-scanning/analyses | jq '.[] | .url'
```
In the response:
find 
```
^"(.+)"$
```
Replace
```
curl -i -L -X DELETE -H "Authorization: Bearer $credentials_GHEC_TOKEN" $1?confirm_delete
```
[Delete a code scanning analysis from a repository](https://docs.github.com/en/rest/code-scanning/code-scanning?apiVersion=2022-11-28#delete-a-code-scanning-analysis-from-a-repository)
```shell
curl -i -L -X DELETE -H "Authorization: Bearer $credentials_GHEC_TOKEN" https://api.github.com/repos/OWNER/REPO/code-scanning/analyses/ANALYSIS_ID?confirm_delete
```
